src/components/text-editor/text-editor.tsx

```diff
@@ -37,60 +13,84 @@ import { getFilteredMenu } from './menu/menu-filter';
+     */
+    @Prop({ reflect: true })
+    public disabled?: boolean;
 
-    @State()
-    private view: EditorView;
+    /**
+     * Set to `true` to make the component read-only.
+     * Use `readonly` when the field is only there to present the data it holds,
+     * and will not become possible for the current user to edit.
+     * :::note
+     * Consider that it might be better to use `limel-markdown`
+     * instead of `limel-text-editor` when the goal is visualizing data.
+     * :::
+     */
+    @Prop({ reflect: true })
+    public readonly?: boolean;
 
     /**
-     * Dispatched when a change is made to the editor
+     * Optional helper text to display below the input field when it has focus
      */
-    @Event()
-    private change: EventEmitter<{ html: string }>;
+    @Prop({ reflect: true })
+    public helperText?: string;
 
-    public componentWillLoad() {}
+    /**
+     * The placeholder text shown inside the input field,
+     * when the field is empty.
+     */
+    @Prop({ reflect: true })
+    public placeholder?: string;
 
-    public render() {
-        return <div id="editor" />;
-    }
+    /**
+     * The label of the editor
+     */
+    @Prop({ reflect: true })
+    public label?: string;
 
-    public componentDidLoad() {
-        const mySchema = new Schema({
-            nodes: addListNodes(schema.spec.nodes, 'paragraph block*', 'block'),
-            marks: schema.spec.marks,
-        });
+    /**
+     * Set to `true` to indicate that the current value of the editor is
+     * invalid.
+     */
+    @Prop({ reflect: true })
+    public invalid?: boolean;
 
-        const menu: MenuElement[][] = buildFullMenu(mySchema)
-            .map((items) => getFilteredMenu(items, this.menuItems))
-            .filter((items) => items.length);
+    /**
+     * Description of the text inside the editor
+     */
+    @Prop({ reflect: true })
+    public value: { html: string };
 
-        this.view = new EditorView(
-            this.host.shadowRoot.querySelector('#editor'),
-            {
-                state: EditorState.create({
-                    doc: DOMParser.fromSchema(mySchema).parse(
-                        this.host.shadowRoot.querySelector('#editor'),
-                    ),
-                    plugins: exampleSetup({
-                        schema: mySchema,
-                        menuContent: menu as MenuItem[][],
-                    }),
-                }),
-                dispatchTransaction: (transaction) => {
-                    const newState = this.view.state.apply(transaction);
-                    this.view.updateState(newState);
+    /**
+     * Dispatched when a change is made to the editor
+     */
+    @Event()
+    public change: EventEmitter<{ html: string }>;
 
-                    this.change.emit({ html: this.view.dom.innerHTML });
-                },
-            },
+    public render() {
+        return (
+            <limel-prosemirror-adapter
```

Comment:
Will this always emit HTML? What about markdown? Markdown takes considerably less space when stored in the database and I think we should default to that instead of HTML. We could have a prop that decides what mode to use if someone really wants HTML

Reply:
@civing What do you think?

Reply:
It would be nice if we stick to one or the other, if we stick to markdown then the consumer could transform the output to HTML of they have a specific need for that? I'm thinking about emails.

Reply:
If we emit markdown, it can be used with the markdown component out of the box to render the content.
If we emit HTML, we don't really have a similar component to use. It should work with the markdown component I think, but we might also unconsciously encourage consumers to use `element.innerHTML` which is really bad

Reply:
Let's take care of this in a follow up PR

Reply:
The limel-markdown component ought to work just fine for sanitising html. But yes, we will probably need to educate people about why they need to run any user generated html through limel-markdown, and not just use `element.innerHTML`.